src/main/java/org/schemaspy/model/Table.java

```diff
@@ -704,6 +705,28 @@ public int compareTo(Table other) {
         return getFullName().compareToIgnoreCase(other.getFullName());
     }
 
+    public boolean hasImpliedConstraints(int degreesOfSeparation) {
+        if(degreesOfSeparation == 0) {
+            return false;
+        }
+
+        for (TableColumn tableColumn : columns.values()) {
+            if(tableColumn.hasImpliedConstraint()) {
+                return true;
+            }
+        }
+        if (degreesOfSeparation > 1) {
+            return columns.values().stream()
+                    .flatMap(tableColumn -> Stream.concat(
+                            tableColumn.getParents().stream(),
+                            tableColumn.getChildren().stream()))
```

Comment:
I had a look and I found a couple of places, they are isolated within DotTableFormatter and there are two type, Tables and Edges (Constraints).

So there might be a point in having immediateRelatives implemented both on TableColumn and on Table. Then there is some ColumnsFilters that at first glance feels semantically incorrect without filter in the name.

I'll create an issue to investigate and look up usage.

Reply:
I've created issue #886 to make a bigger investigation into Table and TableColumns.

Idea is to handle relationships as a whole instead of simply add a method.